src/lib/network_pool/transacation_pool_reorg_tests.ml

```diff
@@ -0,0 +1,433 @@
+  *)
+
+  (*find account in major and minor brnaches with the same nonces and similar balances (less than 100k mina diff)*)
+  let minor_acc_opt = Array.find_map minor ~f:(fun minor_acc -> 
+      Array.find major ~f:(fun major_acc -> 
+            Int.equal minor_acc.nonce major_acc.nonce &&
+            minor_acc.balance - major_acc.balance < 100_000_000_000
+      ) |> Option.map ~f:(fun major_acc -> (major_acc,minor_acc))
+  ) 
+  in
+  let%bind (major_command_spec, minor_command_spec) = match minor_acc_opt with 
+    | Some (major_acc,minor_acc) when limited_capacity -> 
+      
+      let%bind receiver_idx = test_keys |> Array.filter_mapi ~f:(fun i _-> if Int.equal i major_acc.key_idx then None else Some i) |> Quickcheck_lib.of_array in
+      let big_tx_amount = major_acc.balance / 2 in
+      let small_tx_amount = major_acc.balance / 3 in
+      let big_major_tx = Command_spec.Payment { sender = major_acc
+          ; receiver_idx
+          ; fee = minimum_fee
+          ; amount = big_tx_amount
+          } in
+      let new_account_spec = Account_spec.apply_payment big_tx_amount minimum_fee major_acc in 
+      Array.set major new_account_spec.key_idx new_account_spec;
+      (* 3 smaller commands from which first and last one should be dropped *)
+      let minor_txs = Array.init 3 ~f:(fun _ ->
+          let sender = minor.(minor_acc.key_idx) in
+          let small_minor_tx = Command_spec.Payment { sender
+          ; receiver_idx
+          ; fee = minimum_fee
+          ; amount = small_tx_amount - 1_000_000
+          } in
+          let new_account_spec = Account_spec.apply_payment small_tx_amount minimum_fee sender in 
+          Array.set minor minor_acc.key_idx new_account_spec;
+          small_minor_tx
+        )
+         in
+         return (Array.append major_command_spec [|big_major_tx|],
+          Array.append minor_command_spec minor_txs) 
+      
+    | _ -> return (major_command_spec,minor_command_spec)
+  in
+
+  (* Optional Edge Case : Permission Changes:
+
+    - In major sequence, a transaction modifies an account's permissions:
+        1. It removes the permission to maintain the nonce.
+        2. It removes the permission to send transactions.
+    - In minor sequence, there is a regular transaction involving the same account, 
+        but after the permission-modifying transaction in major sequence, 
+        the new transaction becomes invalid and must be dropped.
+  *)
+  let%bind permission_change_cmd = Command_spec.gen_zkapp_blocking_send major in
+  let sender = Command_spec.sender permission_change_cmd in
+  (* We need to increase nonce so transaction has a chance to be placed in the pool.
+    Otherwise it will be dropped as we already have transaction with the same nonce from major sequence
+  *)
+  let sender = major.(sender.key_idx) in
+  let%bind aux_minor_cmd = Command_spec.gen_single_from minor (sender.key_idx,sender) in
```

Comment:
this logic won't imitate the required behavior if `maj1 >= min1`, where `maj1` and `min1` represent the number of txs from `sender` in `major` and `minor` respectively.

I.e. `aux_minor_cmd` will simply be thrown away. We need to add as many aux cmds as necessary to ensure that resulting minor sequence has more transactions from sender than major.